src/core/unit.c

```diff
@@ -1096,13 +1096,16 @@ int unit_add_exec_dependencies(Unit *u, ExecContext *c) {
         }
 
         if (c->private_tmp) {
-                const char *p;
 
-                FOREACH_STRING(p, "/tmp", "/var/tmp") {
-                        r = unit_require_mounts_for(u, p, UNIT_DEPENDENCY_FILE);
-                        if (r < 0)
-                                return r;
-                }
+                /* We support tmp.mount being masked, and this should not be
+                 * considered an error. */
+                r = unit_add_two_dependencies_by_name(u, UNIT_AFTER, UNIT_WANTS, "tmp.mount", true, UNIT_DEPENDENCY_FILE);
+                if (r < 0)
+                        return r;
+
+                r = unit_require_mounts_for(u, "/var/tmp", UNIT_DEPENDENCY_FILE);
```

Comment:
It seems very strange to treat those two directories differently. The setup for private /tmp and private /var/tmp is identical, and the requirements should be identical. Or in other words, the problem with masked /tmp is the same as the problem we'd have with a masked /var or /var/tmp units. I think we just need to handle masked /tmp (or other mounts) more gracefully. But we still want Require deps if the mount is configured.

Reply:
Agreed.